### PR TITLE
protocol-specification: fix preimage encoding

### PR DESCRIPTION
**Pull request by guggero**
Because we ended up encoding the preimage in the Authoriziation
header field as hex, we should also fix the specification. Otherwise
we'll end up with incompatible implementations pretty quickly.

### CODE REVIEW

File: protocol-specification.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ In this chapter, we outline the specification for the abstract LSAT HTTP and gRP
 
 ## Specification
 
-This section defines the "LSAT" authentication scheme, which transmits credentials as `<macaroon(s)>:<preimage>` pairs, where the preimage and the macaroon are each encoded as base64. Multiple macaroons are base64 encoded individually and listed comma separated before the column.  
+This section defines the "LSAT" authentication scheme, which transmits credentials as `<macaroon(s)>:<preimage>` pairs, where the preimage is encoded as hex and the macaroon is encoded as base64. Multiple macaroons are base64 encoded individually and listed comma separated before the column.
```
Comment (wpaulino) [inline protocol-specification.md]: Does our implementation already support handling multiple LSATs? Also, what's the column in `list comma separated before the column`?

Comment (guggero) [reply protocol-specification.md]: No, our implementation doesn't support that yet. But it's required for the discharge macaroons when using 3rd party caveats. So I wanted to have this in the spec to be forward compatible.

Oops, that's supposed to say `colon` not `column`. Fixed.

File: protocol-specification.md [hunk 2]
```diff
@@ -58,19 +58,19 @@ In other words, to receive authorization, the client:
...
 
 ```text
-Authorization: LSAT AGIAJEemVQUTEyNCR0exk7ek90Cg==:MTIzNDEyMzQxMjM0
+Authorization: LSAT AGIAJEemVQUTEyNCR0exk7ek90Cg==:1234abcd1234abcd1234abcd
 ```
 
 ## gRPC Protocol Specification
 
-This section defines the "LSAT" gRPC authentication scheme, which, similarly to the HTTP version, transmits credentials as `<macaroon>:<preimage>` pairs where each field is encoded as base64. As above, this scheme is not considered to be a secure method of user authentication unless used in conjunction with some external secure system such as TLS, as the macaroon and preimage are passed over the network as cleartext.
+This section defines the "LSAT" gRPC authentication scheme, which, similarly to the HTTP version, transmits credentials as `<macaroon(s)>:<preimage>` pairs where the preimage is encoded as hex and the macaroon is encoded as base64. Multiple macaroons are base64 encoded individually and listed comma separated before the column. As above, this scheme is not considered to be a secure method of user authentication unless used in conjunction with some external secure system such as TLS, as the macaroon and preimage are passed over the network as cleartext.
```
Comment (wpaulino) [inline protocol-specification.md]: Should also be fixed here.

**Comment by guggero** [comment]
My fork isn't recognized as a proper fork and for some reason my changes aren't populated correctly. Going to open the same PR again after deleting/forking again.

**Pull request closed without merging**